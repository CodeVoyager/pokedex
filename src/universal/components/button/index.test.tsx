import { shallow } from 'enzyme';
import React from 'react';
import { Button } from '.';

describe('Component', () => {
  describe('Button', () => {
    test("should work on it's own", () => {
      expect(shallow(<Button />)).toMatchSnapshot();
    });
    test('should use specified children', () => {
      expect(shallow(<Button>CONTENT</Button>)).toMatchSnapshot();
      expect(
        shallow(
          <Button>
            <div>CONTENT IN DIV</div>
          </Button>
        )
      ).toMatchSnapshot();
    });
    test('should handle specifing button as mini version', () => {
      expect(shallow(<Button mini />)).toMatchSnapshot();
    });
    test('should run specified onClick event handler', () => {
      const onClick = jest.fn();
      const component = shallow(<Button onClick={onClick} />);

      component.simulate('click');

      expect(onClick).toBeCalled();
      expect(onClick).toBeCalledTimes(1);
    });
  });
});
