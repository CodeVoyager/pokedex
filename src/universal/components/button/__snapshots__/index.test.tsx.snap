// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Button should handle specifing button as mini version 1`] = `
<button
  className="pokemon-button pokemon-button-mini"
/>
`;

exports[`Component Button should use specified children 1`] = `
<button
  className="pokemon-button "
>
  CONTENT
</button>
`;

exports[`Component Button should use specified children 2`] = `
<button
  className="pokemon-button "
>
  <div>
    CONTENT IN DIV
  </div>
</button>
`;

exports[`Component Button should work on it's own 1`] = `
<button
  className="pokemon-button "
/>
`;
